dotnet/src/Connectors/VectorData.Abstractions/RecordOptions/FilteredRecordRetrievalOptions.cs

```diff
@@ -39,76 +38,10 @@ public int Skip
     /// <value>
     /// If not provided, the order of returned results is non-deterministic.
     /// </value>
-    public OrderByDefinition OrderBy { get; } = new();
+    public Func<OrderByDefinition<TRecord>, OrderByDefinition<TRecord>>? OrderBy { get; set; }
```

Comment:
So far, we were using `Expression` rather than `Func` for filters and other similar features. However, every time we did that, we also had to decode the expression to some kind of result that we were interested with.

By making it a `Func` we make it easy to create a builder, just pass it to the method and then work with the results:

```cs
options.OrderBy?.Invoke(new()).Values
```

Moreover, all the APIs that allow for this are public, so any kind of 3rd party connectors can do it as well.

Reply:
Good simplification idea @adamsitnik! Yeah, we only need actual expression trees inside Ascending/Descending, where the property needs to be specified by the user.

(we actually had a very similar thing in EF where we introduced something (ExecuteUpdate) as a big expression tree, and then made a change to have a non-Expression builder pattern with small expression tree fragments, just like here)